### ISSUE TITLE
Inaccessible home dir causes SSL to fallback

### ISSUE THREAD

**Issue by davetrollope** [benoitc/gunicorn#3166]
When using supervisord with the documented example, and Postgres 15 where SSL is enforced, and no certificates are installed, the application fails to connect to the database with gunicorn throwing errors about being unable to read the certificate file, and “no encryption”.

File \"/usr/local/lib/python3.8/site-packages/django/db/backends/base/base.py\", line 200, in connect\n    self.connection = self.get_new_connection(conn_params)\n  File \"/usr/local/lib/python3.8/site-packages/django/utils/asyncio.py\", line 33, in inner\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/site-packages/django/db/backends/postgresql/base.py\", line 187, in get_new_connection\n    connection = Database.connect(**conn_params)\n  File \"/usr/local/lib/python3.8/site-packages/psycopg2/__init__.py\", line 122, in connect\n    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)\npsycopg2.OperationalError: connection to server at \"http://ssl-testing.cbrlwpfnl66e.us-west-2.rds.amazonaws.com\" (10.7.64.199), port 5432 failed: could not open certificate file \"/root/.postgresql/postgresql.crt\": Permission denied\nconnection to server at \"http://ssl-testing.cbrlwpfnl66e.us-west-2.rds.amazonaws.com\" (10.7.64.199), port 5432 failed: FATAL:  no pg_hba.conf entry for host \"10.8.0.89\", user \"rottweiler\", database \"postgres\", no encryption

what I found was, because gunicorn was running as user nobody but it had no home dir, it was unable to read /root as a non priledged user. This causes the permission denied error and causes the SSL connect to fallback to non SSL even though it should be able to connect because their is no certificate at all.

I was able to fix this by simply adding “environment=HOME=/tmp” to the supervisord config.

clearly the right answer to provide a real home dir, but this is a very confusing cause and effect.

I think perhaps the error handling could be better, or it could act as if there is no certificate installed and continue correctly, or at a minimum call out the example doc and ensure the environment includes a valid home dir.

In summary:

Use the basic example supervisord example in a container running as root. 
Do not install certificates.
Hit an endpoint that connects to PG with SSL forced on.

the above error will occur.

**Comment by davetrollope** [comment]
Ok, that’s good info thanks, I did not know it was actually libpq triggering this since I only observed it under gunicorn but in hindsight I realize it is coincidental runtime differences.

I do still think that the example supervisord config for gunicorn could be updated to recommend setting the home dir related to nobody.

Happy to close.
